src/hotspot/share/opto/output.cpp

```diff
@@ -974,6 +974,24 @@ bool PhaseOutput::contains_as_owner(GrowableArray<MonitorValue*> *monarray, Obje
   return false;
 }
 
+// Determine if there is a scalar replaced object description represented by 'ov'.
+bool PhaseOutput::contains_as_scalarized(JVMState* jvms, MachSafePointNode* sfn,
+                                         GrowableArray<ScopeValue*>* objs,
+                                         ObjectValue* ov) const {
+  for (int i = 0; i < jvms->scl_size(); i++) {
+    Node* n = sfn->scalarized(jvms, i);
+    if (!n->is_SafePointScalarObject()) {
```

Comment:
Needs comment explaining what other Node type could be here. May be assert check for them in `else` case.